src/Features/CSharp/Portable/Completion/KeywordRecommenders/DelegateKeywordRecommender.cs

```diff
@@ -31,14 +31,15 @@ public DelegateKeywordRecommender()
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
             return
-                context.IsGlobalStatementContext ||
+                !context.IsGenericConstraintContext &&
+                (context.IsGlobalStatementContext ||
```

Comment:
can you break this into multiple checks.  i find reading and understanding the combination of !, &&, (...) and || rather confusing.